### ISSUE TITLE
remaining_funds field on PreAuthorization is incorrectly flagged as required

### ISSUE THREAD

**Issue by kneirinck** [Mangopay/mangopay2-python-sdk#254]
Since 3.8.2 it seems that creation of a PreAuthorization object now incorrectly requires the remaining_funds property to be present

**Comment by mangomaxoasis** [comment]
We'll lookin into it, shouldn't be mandatory at this stage 
max

**Comment by samitnuk** [comment]
Hi @mangomaxoasis, 

Based on [the doc](https://docs.mangopay.com/guide/check) I believe that `remaining_funds` should be filled by MangoPay and not by MangoPay's customers. Or should we for the versions **3.8.2+** create `PreAuthorization` with `remaining_funds=Money(amount=0, currency='EUR')`?

I have tested `remaining_funds=Money(amount=0, currency='EUR')` with versions **3.8.2** and **3.8.5** and looks like this works fine.

**Comment by mangomaxoasis** [comment]
Hi @kneirinck 
As mentioned by @samitnuk 
The parameter "RemainingFunds" is now associated to PreAuthorizations. 
Since the multi capture [feature](https://docs.mangopay.com/guide/check) , and when this feature is active, several captures can be made against a single PreAuthorization. 
Hence, when you make a calla against a PreAuthorization you get in response a "RemainingFunds" parameter.
Please let le know if this is not clear,
max

**Comment by samitnuk** [comment]
Hi @mangomaxoasis

Thank you for your comment, but I am still not sure what should we do during the `PreAuthorization` creation. I can see two options here:
1.  The `remaining_funds` field is really required on `PreAuthorization` creation - this means that during the `PreAuthorization` creation we should pass some value to it. **So what value we should pass?**
2. The `remaining_funds` field is NOT required on the `PreAuthorization` creation - this means that this is a bug in the SDK.

Please comment or correct me if I missed something.

Because of this issue, we cannot start 3DS2 integration into our platform. 3DS2 requires version **3.8.5** of the SDK.

**Comment by mangomaxoasis** [comment]
Hi @samitnuk 
It's option 2, the field is not passed, it is only in the response and generated by the API 
Can you detail the bug in the SDK?

**Comment by mangomaxoasis** [comment]
Is the SDK forcing you to input this field?

**Comment by samitnuk** [comment]
> Is the SDK forcing you to input this field?

Yes. E.g. when I am doing:
```python
try:
    preauthorization = PreAuthorization(
        author=mangopay_user,
        debited_funds=Money(amount=payment_total, currency='EUR'),
        card=Card.get(card_id),
        secure_mode_return_url=return_url,
    )
    preauthorization.save()
except Exception as e:
    logger.debug(e)
```
I have the next output in the terminal:
```bash
[2021-01-27 18:44:24,063] DEBUG    Missing mandatory field: remaining_funds
```

**Comment by mangomaxoasis** [comment]
@catacraciun Could you take a look ?

**Comment by samitnuk** [comment]
@mangomaxoasis @catacraciun 

Short note - you do not catch this issue with the tests, because in the tests you create `PreAuthorization`s with non-empty `remaining_funds` field.

For example:
https://github.com/Mangopay/mangopay2-python-sdk/blob/aa5cf1f8741a78c697db5a4794b320997d8f4b47/tests/test_preauthorizations.py#L220-L228

**Comment by fredericdelordm** [comment]
Hello @kneirinck,

This has been fixed in our latest release [3.10.1](https://github.com/Mangopay/mangopay2-python-sdk/releases/tag/3.10.1).

Have a nice day,
